crates/pipeline_manager/src/api/connector.rs

```diff
@@ -159,34 +159,34 @@ pub(crate) struct UpdateConnectorResponse {}
             , example = json!(examples::unknown_connector())),
     ),
     params(
-        ("connector_id" = Uuid, Path, description = "Unique connector identifier")
+        ("connector_name" = String, Path, description = "Unique connector name")
     ),
     context_path = "/v0",
     security(("JSON web token (JWT) or API key" = [])),
     tag = "Connectors"
 )]
-#[patch("/connectors/{connector_id}")]
+#[patch("/connectors/{connector_name}")]
 async fn update_connector(
     state: WebData<ServerState>,
     tenant_id: ReqData<TenantId>,
     req: HttpRequest,
     body: web::Json<UpdateConnectorRequest>,
 ) -> Result<HttpResponse, ManagerError> {
-    let connector_id = ConnectorId(parse_uuid_param(&req, "connector_id")?);
-    state
-        .db
-        .lock()
-        .await
-        .update_connector(
-            *tenant_id,
-            connector_id,
-            &body.name,
-            &body.description,
-            &body.config,
-        )
+    let connector_name = parse_string_param(&req, "connector_name")?;
+    let db = state.db.lock().await;
+    let connector = db
+        .get_connector_by_name(*tenant_id, &connector_name)
         .await?;
+    db.update_connector(
```

Comment:
As `db.update_connector` (like all other DB functions) is only used in one point, should the functions themselves not be changed to take the name instead of ID? This will also make it part of the same transaction. This applies to several of the other functions.

It can be done in a next PR, it's more a nice-to-have. Most important is the switch at API level.

Reply:
Yeah, it's been on my mind too. I've been hesitant to migrate the update_* calls because some of them are used in multiple places (e.g. update_program is used via the compiler's interface to the DB too, where it's safer to use IDs). And it felt weird to have some objects be updated by name while others are updated by ID.

I'll save it for a separate PR for now.